src/watchmaker/workers/salt.py

```diff
@@ -89,7 +89,8 @@ def __init__(self, *args, **kwargs):
         # Pop arguments used by SaltBase
         self.user_formulas = kwargs.pop('user_formulas', None) or {}
         self.computer_name = kwargs.pop('computer_name', None) or ''
-        self.ent_env = kwargs.pop('environment', None) or ''
+        self.ent_env = kwargs.pop('environment', None) or None
+        self.valid_envs = kwargs.pop('valid_environments', [None]) or [None]
```

Comment:
The default for `pop` can just be `None`. Then the `or` will takeover. That way the argument default is set really in just the one place rather than two.

Though, it seems to me the default should be an empty list `[]`, not a list containing one item `[None]`.

```suggestion
        self.valid_envs = kwargs.pop('valid_environments', None) or []
```

Reply:
Fixed